src/class_sqlThread.py

```diff
@@ -473,168 +194,249 @@ def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-s
+            upgrade config parser setting version.
+            If the settings version is equal to 2 or 3 then the
+            sqlThread will modify the pubkeys table and change
+            the settings version to 4.
+        """
+        while self.sql_config_settings_version < self.max_setting_level:
+            self.db.upgrade_config_parser_setting_version(self.sql_config_settings_version)
+            self.sql_config_settings_version = self.sql_config_settings_version
+
+    @staticmethod
+    def error_handler(err, command):
+        """
+            Common error handler
+        """
+        if str(err) == 'database or disk is full':
+            logger.fatal(
+                "(While %s) Alert: Your disk or data storage volume is full. sqlThread will now exit.", command
+            )
+            queues.UISignalQueue.put((
+                'alert', (
+                    _translate(
+                        "MainWindow",
+                        "Disk full"),
+                    _translate(
+                        "MainWindow",
+                        'Alert: Your disk or data storage volume is full. Bitmessage will now exit.'),
+                    True)))
+            os._exit(0)
+
+    def is_query_commit(self):
+        """
+            When query == 'commit'
+        """
+        try:
+            self.db.conn.commit()
+        except Exception as err:
+            self.error_handler(err, 'committing')
+
+    def is_query_movemessagstoprog(self):
+        """
+            When query == 'movemessagstoprogs'
+        """
+        logger.debug('the sqlThread is moving the messages.dat file to the local program directory.')
+        try:
+            self.db.conn.commit()
+        except Exception as err:
+            self.error_handler(err, 'movemessagstoprog')
+        self.db.conn.close()
+        shutil.move(
+            paths.lookupAppdataFolder() + 'messages.dat', paths.lookupExeFolder() + 'messages.dat')
+        self.db.conn = sqlite3.connect(paths.lookupExeFolder() + 'messages.dat')
+        self.db.conn.text_factory = str
+        self.db.cur = self.db.conn.cursor()
+
+    def is_query_deleteandvacuume(self):
+        """
+            When query == 'deleteandvacuume'
+        """
+        try:
+            self.db.cur.execute(''' VACUUM ''')
+        except Exception as err:
+            self.error_handler(err, 'deleteandvacuume')
+        self.db.cur.execute('''delete from inbox where folder='trash' ''')
+        self.db.cur.execute('''delete from sent where folder='trash' ''')
+        self.db.conn.commit()
+
+    def is_query_other(self, query):
+        """
+            When the query can be default or other '
+        """
+        parameters = helper_sql.sqlSubmitQueue.get()
+        try:
+            self.db.cur.execute(query, parameters)
+            self.rowcount = self.db.cur.rowcount
+            return self.rowcount
+        except Exception as err:
+            if str(err) != 'database or disk is full':
```

Comment:
replace with `error_handler`

Reply:
`error_handler` is in `else` block.

Reply:
the error handling should be unified, this here isn't a special case

Reply:
Added inside `error_handler`.